inlong-sort/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/MultipleElasticsearchSinkFunctionBase.java

```diff
@@ -0,0 +1,346 @@
+    private final String indexPattern;
+    private final TableSchemaFactory tableSchemaFactory;
+    // initialized and reserved for a later feature.
+    private final SchemaUpdateExceptionPolicy schemaUpdateExceptionPolicy;
+    // open and store an index generator for each new index.
+    private Map<String, IndexGenerator> indexGeneratorMap;
+    // table level metrics
+    private SinkMetricData sinkMetricData;
+    private transient JsonDynamicSchemaFormat jsonDynamicSchemaFormat;
+    private transient SerializationSchema<RowData> serializationSchema;
+
+    public MultipleElasticsearchSinkFunctionBase(
+            @Nullable String docType, // this is deprecated in es 7+
+            SerializationSchema<RowData> serializationSchema,
+            ContentType contentType,
+            RequestFactory<Request, ContentType> requestFactory,
+            Function<RowData, String> createKey,
+            @Nullable Function<RowData, String> createRouting,
+            DirtySinkHelper<Object> dirtySinkHelper,
+            TableSchemaFactory tableSchemaFactory,
+            String multipleFormat,
+            String indexPattern,
+            SchemaUpdateExceptionPolicy schemaUpdateExceptionPolicy) {
+        this.docType = docType;
+        this.serializationSchema = Preconditions.checkNotNull(serializationSchema);
+        this.contentType = Preconditions.checkNotNull(contentType);
+        this.requestFactory = Preconditions.checkNotNull(requestFactory);
+        this.createKey = Preconditions.checkNotNull(createKey);
+        this.createRouting = createRouting;
+        this.dirtySinkHelper = dirtySinkHelper;
+        this.tableSchemaFactory = tableSchemaFactory;
+        this.multipleFormat = multipleFormat;
+        this.indexPattern = indexPattern;
+        this.schemaUpdateExceptionPolicy = schemaUpdateExceptionPolicy;
+    }
+
+    @Override
+    public void open(RuntimeContext ctx, SinkMetricData sinkMetricData) {
+        indexGeneratorMap = new HashMap<>();
+        this.sinkMetricData = sinkMetricData;
+    }
+
+    private void sendMetrics(byte[] document) {
+        if (sinkMetricData != null) {
+            sinkMetricData.invoke(1, document.length);
+        }
+    }
+
+    @Override
+    public void initializeState(FunctionInitializationContext context) {
+    }
+
+    @Override
+    public void snapshotState(FunctionSnapshotContext context) {
+    }
+
+    // dynamically parse the columns of the schema and deserialize the canal-json element
+    private RowData parseElement(RowData element, JsonNode rootNode) {
+        RowType rowType = jsonDynamicSchemaFormat.extractSchema(rootNode);
+        RowData result = jsonDynamicSchemaFormat.extractRowData(rootNode, rowType).get(0);
+        debugconverter(result, rowType);
+        LOGGER.info("parseElement result, rootNode:{}", rootNode);
+        LOGGER.info("parseElement result, rowType:{}", rowType);
+        LOGGER.info("parseElement result, result:{}", result);
```

Comment:
debug ?

Reply:
deleted extra info logs and kept them in dev branch only.